internal/provider/host_resource.go

```diff
@@ -0,0 +1,390 @@
+package provider
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource              = &HostResource{}
+	_ resource.ResourceWithConfigure = &HostResource{}
+)
+
+// NewHostResource is a helper function to simplify the provider implementation
+func NewHostResource() resource.Resource {
+	return &HostResource{}
+}
+
+type HostResourceModelInterface interface {
+	ParseHttpResponse(body []byte) error
+	CreateRequestBody() ([]byte, diag.Diagnostics)
+	GetURL() string
+}
+
+// HostResource is the resource implementation.
+type HostResource struct {
+	client ProviderHTTPClient
+}
+
+// Metadata returns the resource type name.
+func (r *HostResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_host"
+}
+
+// Schema defines the schema for the host resource
+func (r *HostResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+
+			"inventory_id": schema.Int64Attribute{
+				Required: true,
+			},
+			"instance_id": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
```

Comment:
instance_id is computed because, supposing the user does not set it, so, it will be passed as nil, it will be returned by the api as an empty string. Same for description. This is my experience and understanding.

```
{
 ....
  },
  "created": "2024-01-18T16:54:45.198378Z",
  "modified": "2024-01-18T16:54:45.198387Z",
  "name": "tf_host",
  "description": "",
  "inventory": 1,
  "enabled": true,
  "instance_id": "",
  "variables": "",
  "has_active_failures": false,
  "has_inventory_sources": false,
  "last_job": null,
  "last_job_host_summary": null,
  "ansible_facts_modified": null
}
```